client-spec/client-specification-v1.md

```diff
@@ -0,0 +1,322 @@
+# Abstract
+
+This document describes the behaviour that an application must implement
+in order to be a CKAN-compatible mod manager.
+
+Adherence to this document ensures that all mod-management applications
+that consume CKAN metadata behave consistently.
+
+This document refers to version 1 of the CKAN metadata specification.
+
+
+
+# Definitions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in  [RFC 2119](http://www.faqs.org/rfcs/rfc2119.html).
+
+
+
+# Metadata fields
+
+CKAN metadata consists of a JSON file
+conforming to the official JSON schema.
+
+The following sections define how the information contained in the metadata files
+should be interpreted and handled by ckan clients.
+
+
+## Required Fields
+
+The fields described in this section are required by the ckan metadata specification
+and must be present for all valid metadata files.
+A ckan client may assume that all these fields are present in all valid metadata files.
+Clients may assume that all metadata contained in the official CKAN repositories is valid;
+however, client-side validation against the JSON schema is recommended.
+
+### identifier
+
+This is the unique identifier for a module.
+The same identifier is shared by multiple versions of the same module.
+
+A client must use this identifier to discriminate different modules
+and must not allow the user to install two modules with the same identifier.
+
+### version
+
+A string defining the version of the module.
+The specification allows for any string to be used as a version.
+Since there is no standard regarding versioning schemes, a client should not assume any.
+
+As a significant amount of metadata in the official repository
+is obtained automatically from github, leading 'v' characters are common:
+as such, version comparison according to [PEP 440](https://www.python.org/dev/peps/pep-0440/#version-scheme) is recommended.
+Simple alphabetical ordering will often lead to incorrect comparisons
+and is therefore discouraged.
+
+Clients must not allow the installation of multiple versions
+of the same module at the same time.
+
+### download
+
+The download field contains an url where the module can be downloaded.
+This field can contain any url:
+clients must not assume this field to contain a direct link,
+but they must follow eventual redirects.
+
+The download field will always point to a zip archive:
```

Comment:
Actually it does, if you follow the redirect. KS urls have the same behaviour: the url is that of a webpage, but it redirects you to a zip file.